pynance/textimporter.py

```diff
@@ -0,0 +1,205 @@
+import csv
+import datetime
+
+import pandas as pd
+import numpy as np
+
+
+class CsvImporter():
```

Comment:
I propose an interface for `CsvImporter` that is slightly different. Instead of

```py
importer = CsvImporter()
df = importer.read(filename, desc)
```

I'm in favour of
```py
importer = CsvImporter(filename, desc)
df = importer.read() # or even df = importer() by using __call__
```

There are two reasons why I'm in favour:

1. You'll probably never want to do `i = CsvImporter(); df1 = i.read(f1, d1); df2 = i.read(f2,d2)`, i.e. use the same importer for loading two separate files anyways.
1. Designing the interface of the class in a way that makes it single purposed (i.e. it can only load one file) makes reasoning about the class and its class invariant much easier. Currently, `CsvImporter` has no internal state. But lets assume it gets state in future. And other kinds of importers (e.g. those that obtain data online) will have considerable state. If successive calls to `read` get the file parameter, we'd have to make sure that the state does not get "polluted", i.e. that successive calls to `read` still work. But if we were forced by the interface to throw the instance away and make a new one for each import, we'd never have to worry that the class invariant might be broken by `read` since the invariant would be created anew by the constructor for the next import.

Reply:
You're right, reusing the `CsvImporter`does not bring any value.